test/infrastructure/docker/internal/controllers/dockercluster_controller.go

```diff
@@ -153,9 +153,8 @@ func (r *DockerClusterReconciler) reconcileNormal(ctx context.Context, dockerClu
 		return ctrl.Result{}, errors.Wrap(err, "failed to get ip for the load balancer")
 	}
 
-	dockerCluster.Spec.ControlPlaneEndpoint = infrav1.APIEndpoint{
-		Host: lbIP,
-		Port: 6443,
+	if dockerCluster.Spec.ControlPlaneEndpoint.Host == "" {
+		dockerCluster.Spec.ControlPlaneEndpoint.Host = lbIP
```

Comment:
```suggestion
	// Surface the control plane endpoint
	// Note: the control plane port is already set by the user or defaulted by the dockerCluster webhook.
	dockerCluster.Spec.ControlPlaneEndpoint.Host = lbIP
```
So we are reconciling the value continuously